proposals/60-clearing-house-batching-service.md

```diff
@@ -0,0 +1,128 @@
+
+_What must be true for this project to matter?_
+
+* The web3 stack (Filecoin in particular) would be well served by a well defined protocol for batching services that are able to aggregate data and make deals on behalf of end users/developers
+* That definition of such a protocol and reference implementation(s) is not going to cause channel-conflict with existing ecosystem partners who are working in this area (e.g. Powergate) but may enhance their offerings by standardising an interface and workflow for batching
+
+#### User workflow example
+
+_How would a developer or user use this new capability?_
+
+* Create relationship with batching service - signup using DID of some form - get an API key
+* Deposit FIL with the service to be consumed over micro-deals made
+* Submit micro-deals via an API
+  * Either uploading data directly or via IPFS CID
+  * Set maximum FIL for the micro-deal (or accept what the service needs to use)
+  * Set initial replica count
+  * Receive an identifier for the micro-deal to use for later reference
+* Check micro-deal status via API
+
+An MVP would trim this down to remove user identification, FIL handling and replicas.
+
+#### Impact
+
+_How would this directly contribute to web3 dev stack product-market fit?_
+
+#### Leverage
+
+_How much would nailing this project improve our knowledge and ability to execute future projects?_
+
+#### Confidence
+
+_How sure are we that this impact would be realized? Label from [this scale](https://medium.com/@nimay/inside-product-introduction-to-feature-priority-using-ice-impact-confidence-ease-and-gist-5180434e5b15)_.
+
+## Project definition
+
+#### Brief plan of attack
+
+<!--Briefly describe the milestones/steps/work needed for this project-->
+
+#### What does done look like?
+
+_What specific deliverables should completed to consider this project done?_
+
+####  What does success look like?
+
+_Success means impact. How will we know we did the right thing?_
+
+#### Counterpoints &amp; pre-mortem
+
+_Why might this project be lower impact than expected? How could this project fail to complete, or fail to be successful?_
+
+* This places an intermediate layer between miner and client which introduces potential safety and reputational concerns. A miner can't do a plain filter by CID in this mechanism since the actual data CID will be buried inside the deal (unless Lotus looks for filtered CIDs deep inside the DAG?). A batching service instance would have to have a relationship with its miners and would offload some of the filtering requirements. So miners needing baseline guarantees about content would need to ensure the batching service they deal with imposes that on clients.
+* There may be concerns with adjacent content inside single deals if your content is places next to objectionable content. I'm not sure how this plays out, but CID+selector where you manually adjusting the selector slightly gives you someone else's content might have unfortunate side-effects?
```

Comment:
This might be a good q for the Murmuration folks, but I believe for things like Bitscreen they're basically adding content to a do not retrieve list -> which maybe @ribasushi  has some recollection if this could be easily altered to support CID+selector?

Reply:
Um... selectors apply during retrieval alone. I think @rvagg was more concerned "my content is in the same PieceCID as this other stuff" being more problematic than "my deal for content is in a sector with other deals for that other stuff"